compiler/src/dmd/funcsem.d

```diff
@@ -1931,6 +1931,112 @@ FuncDeclaration overloadExactMatch(FuncDeclaration thisfd, Type t)
     return fd;
 }
 
+/****************************************************
+ * Determine if fd1 overrides fd2.
+ * Return !=0 if it does.
+ */
+extern (D) int overrides(FuncDeclaration fd1, FuncDeclaration fd2)
+{
+    int result = 0;
+    if (fd1.ident == fd2.ident)
+    {
+        const cov = fd1.type.covariant(fd2.type);
+        if (cov != Covariant.distinct)
+        {
+            ClassDeclaration cd1 = fd1.toParent().isClassDeclaration();
+            ClassDeclaration cd2 = fd2.toParent().isClassDeclaration();
+            if (cd1 && cd2 && cd2.isBaseOf(cd1, null))
+                result = 1;
+        }
+    }
+    return result;
+}
+
+/*************************************
+ * Determine partial specialization order of functions `f` vs `g`.
+ * This is very similar to TemplateDeclaration::leastAsSpecialized().
+ * Params:
+ *  f = first function
+ *  g = second function
+ *  names = names of parameters
+ * Returns:
+ *      match   'this' is at least as specialized as g
+ *      0       g is more specialized than 'this'
+ */
+MATCH leastAsSpecialized(FuncDeclaration f, FuncDeclaration g, Identifiers* names)
```

Comment:
You'll need to add this to the dmd namespace in [cxxfrontend.d](https://github.com/dlang/dmd/blob/e4eae5990c10dfa030a7cb126cb3099791aaeb73/compiler/src/dmd/cxxfrontend.d) for C++ backends.

Then [move](https://github.com/dlang/dmd/blob/e4eae5990c10dfa030a7cb126cb3099791aaeb73/compiler/src/dmd/declaration.h#L709) the declaration [here](https://github.com/dlang/dmd/blob/e4eae5990c10dfa030a7cb126cb3099791aaeb73/compiler/src/dmd/declaration.h#L36).

Reply:
Cheers, done.